includes/gateways/paypal/webhooks/events/class-payment-capture-completed.php

```diff
@@ -40,14 +43,14 @@ protected function process_event() {
 			throw new \Exception( 'Missing amount value.', 200 );
 		}
 
-		if ( ! isset( $this->event->resource->amount->currency_code ) || strtoupper( $this->event->resource->amount->currency_code ) !== strtoupper( $payment->currency ) ) {
-			throw new \Exception( sprintf( 'Missing or invalid currency code. Expected: %s; PayPal: %s', $payment->currency, esc_html( $this->event->resource->amount->currency_code ) ), 200 );
+		if ( ! isset( $this->event->resource->amount->currency_code ) || strtoupper( $this->event->resource->amount->currency_code ) !== strtoupper( $order->currency ) ) {
+			throw new \Exception( sprintf( 'Missing or invalid currency code. Expected: %s; PayPal: %s', $order->currency, esc_html( $this->event->resource->amount->currency_code ) ), 200 );
 		}
 
-		$paypal_amount  = (float) $this->event->resource->amount->value;
-		$payment_amount = edd_get_payment_amount( $payment->ID );
+		$paypal_amount = (float) $this->event->resource->amount->value;
+		$order_amount  = edd_get_payment_amount( $order->id );
```

Comment:
Should this be `$order->total` instead of using the payment function? Or I guess it's okay since it's using the order object and running through a filter.

Reply:
I used the payment function only because it has the filter in place. (I hate that filter.)